packages/resources-provider-plugin/src/index.ts

```diff
@@ -182,6 +184,60 @@ module.exports = (server: ResourceProviderApp): Plugin => {
     server.setPluginStatus(msg)
   }
 
+  /** process changes in config schema */
+  const cleanConfig = (options: ProviderSettings): ProviderSettings => {
+    server.debug(`Check / Clean loaded settings...`)
+
+    const defaultConfig: ProviderSettings = {
+      standard: {
+        routes: true,
+        waypoints: true,
+        notes: true,
+        regions: true,
+        charts: true
+      },
+      custom: []
+    }
+
+    // set / save defaults if no saved settings
+    if (!options?.standard) {
+      server.savePluginOptions(defaultConfig, () => {
+        server.debug(`Default configuration applied...`)
+      })
+      return defaultConfig
+    }
+
+    let isCleaned = false
+    // check / clean settings
+    if (!Array.isArray(options?.custom)) {
+      options.custom = []
+      isCleaned = true
+    }
```

Comment:
Why do we need the extra logic for charts and not for the other resource types? I am guessing because charts is added and not there earlier? If that is the case I would write the code so that it works for all the standard resource types and charts is not an exception that requires special handling. That would be easier to understand looking back at the code in the future. And future proof, in case we want to add something else.

Reply:
It's there to catch if charts was defined as a custom resource and update the settings accordingly.

It shouldn't apply to the other standard resource types as they have always been a check box.